src/test/java/net/snowflake/ingest/streaming/internal/ParquetBufferTest.java

```diff
@@ -0,0 +1,278 @@
+
+import net.snowflake.ingest.streaming.OpenChannelRequest;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class ParquetBufferTest {
+  private ParquetRowBuffer rowBufferOnErrorContinue;
+
+  @Before
+  public void setupRowBuffer() {
+    this.rowBufferOnErrorContinue = createTestBuffer(OpenChannelRequest.OnErrorOption.CONTINUE);
+    this.rowBufferOnErrorContinue.setupSchema(RowBufferTest.createSchema());
+  }
+
+  ParquetRowBuffer createTestBuffer(OpenChannelRequest.OnErrorOption onErrorOption) {
+    SnowflakeStreamingIngestClientInternal<ParquetChunkData> client =
+        new SnowflakeStreamingIngestClientInternal<>("client");
+    SnowflakeStreamingIngestChannelInternal<ParquetChunkData> channel =
+        new SnowflakeStreamingIngestChannelInternal<>(
+            "channel", "db", "schema", "table", "0", 0L, 0L, client, "key", 1234L, onErrorOption);
+
+    return new ParquetRowBuffer(channel);
+  }
+
+  @Test
+  public void buildFieldFixedSB1() {
+    // FIXED, SB1
+    ColumnMetadata testCol = RowBufferTest.generateCol("SB1", "FIXED", true);
+    Type result = this.rowBufferOnErrorContinue.getColumnParquetType(testCol, 0);
+
+    Assert.assertEquals("testCol", result.getName());
+    Assert.assertEquals(4, result.asPrimitiveType().getTypeLength());
+    Assert.assertEquals(0, result.asPrimitiveType().getId().intValue());
+
+    Assert.assertEquals(
+        PrimitiveType.PrimitiveTypeName.INT32, result.asPrimitiveType().getPrimitiveTypeName());
+    Assert.assertEquals(
+        LogicalTypeAnnotation.decimalType(testCol.getScale(), testCol.getPrecision()),
+        result.getLogicalTypeAnnotation());
+    Assert.assertEquals(Type.Repetition.OPTIONAL, result.getRepetition());
+  }
+
+  @Test
+  public void buildFieldFixedSB2() {
+    ColumnMetadata testCol = RowBufferTest.generateCol("SB2", "FIXED", false);
+    Type result = this.rowBufferOnErrorContinue.getColumnParquetType(testCol, 0);
+
+    Assert.assertEquals("testCol", result.getName());
+    Assert.assertEquals(4, result.asPrimitiveType().getTypeLength());
+    Assert.assertEquals(0, result.asPrimitiveType().getId().intValue());
+
+    Assert.assertEquals(
+        PrimitiveType.PrimitiveTypeName.INT32, result.asPrimitiveType().getPrimitiveTypeName());
+    Assert.assertEquals(
+        LogicalTypeAnnotation.decimalType(testCol.getScale(), testCol.getPrecision()),
+        result.getLogicalTypeAnnotation());
+    Assert.assertEquals(Type.Repetition.REQUIRED, result.getRepetition());
+  }
+
+  @Test
+  public void buildFieldFixedSB4() {
+    ColumnMetadata testCol = RowBufferTest.generateCol("SB4", "FIXED", true);
+    Type result = this.rowBufferOnErrorContinue.getColumnParquetType(testCol, 0);
+
+    Assert.assertEquals("testCol", result.getName());
```

Comment:
We also implicitly pack physical/logical types into `metadata` in `getColumnParquetType`.
This needs to be tested as well. If you consider a refactoring into a `ParquetTypeGenerator`,
It could return `type` + `metadata` something to test.

Reply:
done.